server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/MergedPointRangeQuery.java

```diff
@@ -0,0 +1,201 @@
+                StringBuilder sb = new StringBuilder();
+                sb.append("(");
+                for (int i = 0; i < value.length; i++) {
+                    if (i > 0) {
+                        sb.append(' ');
+                    }
+                    sb.append(Integer.toHexString(value[i] & 0xFF));
+                }
+                sb.append(')');
+                return sb.toString();
+            }
+        };
+    }
+
+    @Override
+    public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+        Weight delegateForMultiValuedSegmentsWeight = delegateForMultiValuedSegments.createWeight(searcher, scoreMode, boost);
+        return new Weight(this) {
+            Weight mostCompactWeight;
+
+            @Override
+            public boolean isCacheable(LeafReaderContext ctx) {
+                return delegateForMultiValuedSegmentsWeight.isCacheable(ctx);
+            }
+
+            @Override
+            public Scorer scorer(LeafReaderContext context) throws IOException {
+                ScorerSupplier scorerSupplier = scorerSupplier(context);
+                if (scorerSupplier == null) {
+                    return null;
+                }
+                return scorerSupplier.get(Long.MAX_VALUE);
+            }
+
+            @Override
+            public ScorerSupplier scorerSupplier(LeafReaderContext context) throws IOException {
+                /*
+                 * If we're sure docs only have a single value for the field
+                 * we can pick the most compact bounds. If there are multiple values
+                 * for the field we have to run the boolean query.
+                 */
+                PointValues points = context.reader().getPointValues(field);
+                if (points == null) {
+                    return null;
+                }
+                if (points.size() == points.getDocCount()) {
+                    // Each doc that has points has exactly one point.
+                    if (mostCompactWeight == null) {
+                        mostCompactWeight = delegateForSingleValuedSegments.createWeight(searcher, scoreMode, boost);
+                    }
+                    return mostCompactWeight.scorerSupplier(context);
+                }
+                return delegateForMultiValuedSegmentsWeight.scorerSupplier(context);
+            }
+
+            @Override
+            public BulkScorer bulkScorer(LeafReaderContext context) throws IOException {
+                return super.bulkScorer(context);
+            }
+
+            @Override
+            @Deprecated
+            public void extractTerms(Set<Term> terms) {
+                delegateForMultiValuedSegmentsWeight.extractTerms(terms);
+            }
+
+            @Override
+            public Explanation explain(LeafReaderContext context, int doc) throws IOException {
+                return delegateForMultiValuedSegmentsWeight.explain(context, doc);
+            }
```

Comment:
We can safely delegate this method to the default implementation?

Reply:
Sorry, I don't understand which default implementation you mean.

Reply:
I meant that the parent class has already an implementation and we do not need to override it. In any case I think this method is only use when called explain and in this case, it will never be used?

Reply:
I talked to @iverase and he thought I was using `ConstantScoreWeight`. I probably should have been using it. So I am now.